### PR TITLE
Update and restructure the Editor Hooks doc.

### PR DESCRIPTION
**Pull request by ndiego**
## What?
This PR updates the [Editor Hooks](https://developer.wordpress.org/block-editor/reference-guides/filters/editor-filters/) doc and does a bit of restructuring. 

## Why?
In my work to update/enhance the [Curating the Editor Experience](https://github.com/WordPress/gutenberg/issues/56170) documentation, I have been identifying other areas of the documentation that are related and need updating. This Editor Hooks doc could use more examples and some restructuring. I updated it to place greater emphasis on `block_editor_settings_all` and a few additional examples of how to use the hook. Rather than completely recreate these examples in the [Curating the Editor Experience](https://developer.wordpress.org/block-editor/how-to-guides/curating-the-editor-experience/) section, it's better to improve the source doc and then link to it.

I would consider this PR a "first pass". There is probably a lot more we can add to the Editor Hooks doc.

### CODE REVIEW

File: docs/reference-guides/filters/editor-filters.md [hunk 1]
```diff
@@ -1,23 +1,166 @@
...
 
-To modify the behavior of the editor experience, WordPress exposes several APIs.
+WordPress exposes several APIs that allow you to modify the editor experience.
+
+## Editor settings
+
+One of the most common ways to modify the Editor is through the [`block_editor_settings_all`](https://developer.wordpress.org/reference/hooks/block_editor_settings_all/) PHP filter, which is applied before settings are sent to the initialized Editor. This filter allows plugin and theme authors extensive control over how the Editor behaves.
+
+<div class="callout callout-warning">
+	Before WordPress 5.8, this hook was known as <code>block_editor_settings</code>, which is now deprecated. If you need to support older versions of WordPress, you might need a way to detect which filter should be used. The recommended way to proceed is to check if the <code>WP_Block_Editor_Context</code> class exists.
```
Comment (juanmaguitar) [inline docs/reference-guides/filters/editor-filters.md]: > The recommended way to proceed is to check if the <code>WP_Block_Editor_Context</code> class exists.

In which way checking if the <code>WP_Block_Editor_Context</code> class exists, does allow to determine which filter should be used? If it does exist then [block_editor_settings_all](https://developer.wordpress.org/reference/hooks/block_editor_settings_all/) should be used? 

Maybe we could be more clear about this here.

**Comment by juanmaguitar** [comment]
Would it make sense to create an example at [block-development-examples](https://github.com/WordPress/block-development-examples) with all these restrictions/enhancements and provide a link to it so these code examples can be seen in action?

If you agree, for the sake of this PR, it would be enough to create [the issue](https://github.com/WordPress/block-development-examples/issues) for this new example.

File: docs/reference-guides/filters/editor-filters.md [hunk 2]
```diff
@@ -66,112 +209,38 @@ addFilter(
...
 
 ```js
+import { addAction } from '@wordpress/hooks';
+
 addAction(
 	'editor.ErrorBoundary.errorLogged',
 	'mu-plugin/error-capture-setup',
 	( error ) => {
-		// error is the exception's error object
+		// Error is the exception's error object.
```
Comment (juanmaguitar) [inline docs/reference-guides/filters/editor-filters.md]: I understand that `ErrorCaptureTool.captureError` is an example of a (non-existing) tool that would track the error. Still, this example wouldn't work when copied and pasted to a real project.
I would be more explicit about this or would shape the example to replace `ErrorCaptureTool.captureError( error );` by `console.log( error );`

> You can use this action to get hold of the error object handled by the boundaries. For example, you may want to send them to an external error-tracking tool. Here's an example:

```
addAction(
	'editor.ErrorBoundary.errorLogged',
	'mu-plugin/error-capture-setup',
	( error ) => {
		// error is the exception's error object. 
		// we can console.log it or send it to an external error-tracking tool
		console.log ( error );
	}
);
```

**Review by juanmaguitar — APPROVED** [review]

**Review by juanmaguitar — APPROVED** [review]

File: docs/reference-guides/filters/editor-filters.md (hunk 1 above)
Comment (ndiego) [reply docs/reference-guides/filters/editor-filters.md]: Good call, I'll update. `WP_Block_Editor_Context` was introduced in 5.8, so it exists, you are safe to use `block_editor_settings_all`.

File: docs/reference-guides/filters/editor-filters.md (hunk 2 above)
Comment (ndiego) [reply docs/reference-guides/filters/editor-filters.md]: Good call. I update this in the most recent commit.

**Pull request merged**